File: Latihan2_2/python/Memory.py
from Hardware import Hardware

# class Memory adalah child dari class Hardware
class Memory(Hardware):
    # atribut private
    __frequency = ""
    __memorySize = ""
    __supportsCuda = ""

    # constructor
    def __init__(self):
        self.__frequency = ""
        self.__memorySize = ""
        self.supportsCuda = ""

    # set memory frequency
    def set_frequency(self, frequency):
        self.__frequency = frequency
    # set memory frequency
    def get_frequency(self):
        return self.__frequency

    # set memory memorySize
    def set_memorySize(self, memorySize):
        self.__memorySize = memorySize
    # set memory memorySize
    def get_memorySize(self):
        return self.__memorySize

    # set memory supportsCuda
    def set_supportsCuda(self, supportsCuda):
        self.__supportsCuda = supportsCuda
    # set memory supportsCuda
    def get_supportsCuda(self):
        return self.__supportsCuda

File: Latihan2_2/cpp/Memory.cpp
#include "Hardware.cpp"

// class Memory adalah child dari class Hardware
class Memory: public Hardware
{
    // atribut private
    private:
        string frequency;
        string memorySize;
        string supportsCuda;
        
    public:
        // constructor
        Memory(string frequency, string memorySize, string supportsCuda) {
            this->frequency = frequency;
            this->memorySize = memorySize;
            this->supportsCuda = supportsCuda;
        }

        // constructor kosong
        Memory() {
        }

        // set memory frequency
        void set_frequency(string frequency) {
            this->frequency = frequency;
        }
        // get memory frequency
        string get_frequency() {
            return this->frequency;
        }

        // set memory memorySize
        void set_memorySize(string memorySize) {
            this->memorySize = memorySize;
        }
        // get memory memorySize
        string get_memorySize() {
            return this->memorySize;
        }
        
        // set memory SupportCuda
        void set_supportsCuda(string supportsCuda) {
            this->supportsCuda = supportsCuda;
        }
        // get memory supportCuda
        string get_supportsCuda() {
            return this->supportsCuda;
        }

        // destructor
        ~Memory() {
        }
};